test/e2e/apps/deployment.go

```diff
@@ -880,12 +884,19 @@ func testProportionalScalingDeployment(f *framework.Framework) {
 	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), secondRS)
 	Expect(err).NotTo(HaveOccurred())
 
-	framework.Logf("Verifying that first rollout's replicaset has 11 available replicas")
-	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, 11)
+	numReplicaDiff := newReplicas - replicas
+	if numReplicaDiff < 0 {
+		numReplicaDiff = -numReplicaDiff
+	}
+	biggerProportion := (numReplicaDiff + 1) / 2
+	availableReplicas = availableReplicas + biggerProportion
+	framework.Logf("Verifying that first rollout's replicaset has %d available replicas", availableReplicas)
+	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, availableReplicas)
 	Expect(err).NotTo(HaveOccurred())
 
-	framework.Logf("Verifying that second rollout's replicaset does not have any available replicas")
-	Expect(secondRS.Status.AvailableReplicas).Should(Equal(int32(0)))
+	availableReplicas = numReplicaDiff*2 - biggerProportion
```

Comment:
Note that smaller proportion can be obtained from numReplicaDiff - biggerProportion. As we have numReplicaDiff + (numReplicaDiff - biggerProportion), it can be simplified as numReplicaDiff*2 - biggerProportion.